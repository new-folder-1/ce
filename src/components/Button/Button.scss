.Button {
    height: 30px;
    
    border-radius: 2px;
    font-weight: bold;
    color: #fff;

    &[disabled] {
        opacity: 0.5;
    }

    &_theme_default {
        border: 1px solid #fff;
        background-color: #255dd3;
    }
    &_theme_action {
        border: none;
        background-color: #f57c00;
    }
}
