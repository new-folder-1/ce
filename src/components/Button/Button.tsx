import * as React from 'react';

import './Button.scss';

interface ButtonProps {
    text: string;
    disabled?: boolean;
    onClick?: () => void;
    className?: string;
    theme?: ButtonTheme;
}

type ButtonTheme = 'default' | 'action';

export const Button = React.memo(({ text, theme, className, ...rest }: ButtonProps) => { 
    const bClassName = `Button Button_theme_${theme || 'default'} ${className || ''}`;
    return (
        <button
            className={bClassName}
            {...rest}
        >
            {text}
        </button>
    );
});
