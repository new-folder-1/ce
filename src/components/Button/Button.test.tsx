import React from 'react';
import { shallow } from 'enzyme';

import { Button } from './Button';

describe('Button', () => {
    it('renders with default theme if one not provided', () => {
        expect(
            shallow(<Button text="test" />)
        ).toMatchSnapshot();
    });

    it('renders with action theme', () => {
        expect(
            shallow(<Button text="test" theme="action" />)
        ).toMatchSnapshot();
    });

    it('renders disabled', () => {
        expect(
            shallow(<Button text="test" disabled />)
        ).toMatchSnapshot();
    });

    it('renders with className added', () => {
        expect(
            shallow(<Button text="test" className="Wrapper-Button" />)
        ).toMatchSnapshot();
    });

    it('onClick fires', () => {
        const onClick = jest.fn()
        const component = shallow(
            <Button text="test" className="Wrapper-Button" onClick={onClick} />
        );
        component.find('button').simulate('click');
        
        expect(onClick).toBeCalledTimes(1);
    });
});
