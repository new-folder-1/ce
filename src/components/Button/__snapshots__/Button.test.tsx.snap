// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders disabled 1`] = `
<button
  className="Button Button_theme_default "
  disabled={true}
>
  test
</button>
`;

exports[`Button renders with action theme 1`] = `
<button
  className="Button Button_theme_action "
>
  test
</button>
`;

exports[`Button renders with className added 1`] = `
<button
  className="Button Button_theme_default Wrapper-Button"
>
  test
</button>
`;

exports[`Button renders with default theme if one not provided 1`] = `
<button
  className="Button Button_theme_default "
>
  test
</button>
`;
